schema_based_on_diagram.sql

```diff
@@ -1,3 +1,73 @@
 -- make changes 
 -- add gitflow
--- add README
\ No newline at end of file
+-- add README
+
+-- create patients table :
+
+CREATE TABLE patients (
+    id INT PRIMARY KEY,
+    name VARCHAR(255),
+    date_of_birth DATE
+);
+
+-- create medical_histories table and add foreign key
+
+CREATE TABLE medical_histories (
+    id INT PRIMARY KEY,
+    admitted_at TIMESTAMP,
+    patient_id INT,
+    status VARCHAR(255)
+    -- CONSTRAINT fk_medical_histories FOREIGN KEY (patient_id) REFERENCES patients(id)
+);
+
+-- create the "treatments" table with the "id" column as a foreign key referencing the "id" column in the "medical_histories" table :
+
+CREATE TABLE treatments (
+    id INT PRIMARY KEY,
+    type VARCHAR(255),
+    name VARCHAR(255)
+    -- CONSTRAINT fk_treatments_medical_histories FOREIGN KEY (id) REFERENCES medical_histories(id)
+);
+
+--join table
+CREATE TABLE medical_histories_has_treatments (
+    medical_history_id INT,
+    treatment_id INT,
+    CONSTRAINT fk_mhht_medical_histories FOREIGN KEY (medical_history_id) REFERENCES medical_histories(id),
+    CONSTRAINT fk_mhht_treatments FOREIGN KEY (treatment_id) REFERENCES treatments(id)
+);
+
+-- invoice_items table with all columns and foreign key referneced :
+CREATE TABLE invoice_items (
+    id INT PRIMARY KEY,
+    unit_price DECIMAL(10, 2),
+    quantity INT,
+    total_price DECIMAL(10, 2),
+    invoice_id INT,
+    treatment_id INT,
+    CONSTRAINT fk_invoice_items_treatments FOREIGN KEY (treatment_id) REFERENCES treatments(id)
+);
+
+ CREATE TABLE invoices (
+    id INT PRIMARY KEY,
+    total_amount DECIMAL(10, 2),
+    generated_at TIMESTAMP,
+    payed_at TIMESTAMP,
+    medical_history_id INT,
+    CONSTRAINT fk_invoices_medical_histories FOREIGN KEY (medical_history_id) REFERENCES medical_histories(id),
+);
+
+ALTER TABLE invoice_items
+ADD CONSTRAINT fk_invoices_invoiceItems FOREIGN KEY (invoice_id) REFERENCES invoices(id);
+
+-- Add a B-tree index to the "patient_id" foreign key in the "medical_histories" table
+CREATE INDEX idx_patient_id ON medical_histories(patient_id);
+
+-- Add a B-tree index to the "id" foreign key in the "treatments" table
+CREATE INDEX idx_medical_history_id ON treatments(id);
```

Comment:
👍 Great work so far! However, I noticed a couple of points to consider:

- [ ] As per the project requirements, foreign keys should be indexed to enhance database performance. While you've added indexes for some of the foreign keys, the `treatments` table's `id` column is not a foreign key; it's the primary key. Kindly review your code and ensure that you're adding indexes to the foreign keys as required.

- [ ] Additionally, remember to check your entire database for foreign keys that may be missing indexes and add them as needed to optimize database performance across all relevant tables.

By following these steps, you'll ensure that your database is properly indexed, leading to improved query performance. Keep up the good work! 👍🗃️